README.md

```diff
@@ -184,14 +189,149 @@ Binaries built by Burrito include a built-in set of commands for performing main
 
 * `./my-binary maintenance uninstall` - Will prompt to uninstall the unpacked payload on the host machine.
 
+## Advanced Build Configuration
+
+#### Build Steps and Phases
+
+Burrito runs the mix release task in three "Phases". Each of these phases contains a number of "Steps", and a context struct containing the current state of the build, which is passed between each step.
+
+The three phases of the Burrito build pipeline are:
+
+  * `Fetch` - This phase is responsible for downloading or copying in any replacement ERTS builds for cross-build targets.
+  * `Patch` - The patch phase injects custom scripts into the build directory, this phase is also where any custom files should be copied into the build directory before being archived.
+  * `Build` -  This is the final phase in the build flow, it produces the final wrapper binary with a payload embedded inside.
+  
+  You can add your own steps before and after phases execute. Your custom steps will also receive the build context struct, and can return a modified one to customize a build to your liking.
+
+  An example of adding a step before the fetch phase, and after the build phase:
+
+  ```elixir
+  # ... mix.exs file
+  def releases do
+    [
+      my_app: [
+        steps: [:assemble, &Burrito.wrap/1],
+        burrito: [
+          # ... other Burrito configuration
+          extra_steps: [
+            fetch: [pre: [MyCustomStepModule, AnotherCustomStepModule]],
+            build: [post: [CustomStepAgain, YetAnotherCustomStepModule]]
+            # ...
+          ]
+        ]
+      ]
+    ]
+  end
+  # ...
+  ```
+
+#### Build Targets and Qualifiers
+
+A burrito build target is a keyword list that contains an operating system, a CPU architecture, and extra build options (called Qualifiers).
+
+Here's a build target that defines Linux, 64-bit x86:
+
+```elixir
+  def releases do
+  [
+    example_cli_app: [
+      steps: [:assemble, &Burrito.wrap/1],
+      burrito: [
+        targets: [
+          linux: [os: :linux, cpu: :x86_64],
+        ],
+      ]
+    ]
+  ]
+  end
+```
+
+If we want to define another build target for linux, but with Musl libc instead of GNU libc we can add a Build Qualifier.
```

Comment:
The musl project stylizes themselves as "musl", and "glibc" or "GNU C Library" would be more recognizable ways of writing "GNU libc".

Maybe this change reads more clearly?

> Build targets can be further customized using build qualifiers. For example, a Linux build target can be configured to use `musl` instead of `glibc` using the following definition: